### PR TITLE
Output test results in junit and export to circle

### PR DESCRIPTION
**Pull request by cpuguy83**
This adds an override for `go test` so in CI we can use `gotestsum` which outputs junit formatted results which will be consumed by circleci.

### CODE REVIEW

**Comment by cpuguy83** [comment]
Example test insights report: https://app.circleci.com/insights/github/virtual-kubelet/virtual-kubelet/workflows/validate_and_test/tests?branch=pull/973

Also failure reports, much better than sifting through the logs:

Job list view:

image

Detail view:

image
image

File: .circleci/config.yml [hunk 1]
```diff
@@ -42,13 +42,20 @@ jobs:
...
           command: V=1 make build
+      - run: make gotestsum
       - run:
           name: Tests
-          command: V=1 CI=1 make test envtest
+          environment:
+            GOTEST: gotestsum -- -timeout=9m
+            GOTESTSUM_JUNITFILE: output/unit/results.xml
+            GODEBUG: cgocheck=2
+          command: mkdir -p output/unit; V=1 make test envtest
```
Comment (sargun) [inline .circleci/config.yml:52]: why not make this multiline as well?

File: .circleci/config.yml [hunk 2]
```diff
@@ -121,13 +128,21 @@ jobs:
...
+            GOTEST: gotestsum --
           command: |
             mkdir $HOME/.go
             export PATH=$HOME/.go/bin:${PATH}
             curl -fsSL -o "/tmp/go.tar.gz" "https://dl.google.com/go/go1.15.6.linux-amd64.tar.gz"
             tar -C $HOME/.go --strip-components=1 -xzf "/tmp/go.tar.gz"
             go version
+            mkdir -p output/e2e
+            export GOTESTSUM_JUNITFILE="$(pwd)/output/e2e/results.xml"
+            export PATH="${GOPATH}/bin:${PATH}"
```
Comment (sargun) [inline .circleci/config.yml:141]: Any reason not to use gobin?

File: Makefile [hunk 3]
```diff
@@ -196,3 +186,6 @@ fmt:
 .PHONY: lint
 lint: $(gobin_tool)
 	gobin -run github.com/golangci/golangci-lint/cmd/golangci-lint@v1.33.0 run ./...
+
+gotestsum: $(gobin_tool)
+	gobin gotest.tools/gotestsum@v1.6.4
```
Comment (sargun) [inline Makefile:191]: shouldn't it just be:
`gobin -run gotest.tools/gotestsum@v1.6.4 -- ` -- versus installing it into the path?

**Review by sargun — COMMENTED** [review]
Go 1.17 is supposed to move everything to `go run`:
https://github.com/golang/go/issues/42088

File: .circleci/config.yml (hunk 1 above)
Comment (cpuguy83) [reply .circleci/config.yml:52]: Just because it's short.

File: Makefile (hunk 3 above)
Comment (cpuguy83) [reply Makefile:191]: Then I have to re-execute the Makefile to run gotestsum.

File: .circleci/config.yml (hunk 2 above)
Comment (cpuguy83) [reply .circleci/config.yml:141]: As in `gobin -run`?

Comment (sargun) [reply .circleci/config.yml:143]: Yep

Comment (cpuguy83) [reply .circleci/config.yml:143]: I'm just going to update to go1.16 and use `go install`

Comment (cpuguy83) [reply .circleci/config.yml:143]: *In the circle config*

**Review by pires — APPROVED** [review]
Nice!

**Comment by cpuguy83** [comment]
@sargun Are you ok with these changes?

**Pull request merged**